### ISSUE TITLE
Training Code Available?

### ISSUE THREAD

**Issue by omkar-kumbhar** [cjspoerer/rcnn-sat#1]
Is there a training code available? I tried coming up with my own training method by referring to the code and paper. But I think I'm missing out on some details. 

I'm trying to retrain the model on CIFAR10, and while it's mentioned the BL was trained with cumulative_readout=False, I trained it on the flag set to 'True'.

Here are the training steps I followed:
1. Converted CIFAR10 32x32 images to match the network input of 128x128 by upscaling. 
2. Defined model as given in the description:
>>    input_layer = tf.keras.layers.Input((128, 128, 3))
>>    model = bl_net(input_layer, classes=10, cumulative_readout=True)
3. Using the loss and optimizer explained in the paper:
>>    cce = tf.keras.losses.CategoricalCrossentropy()
>>    opt = tf.keras.optimizers.Adam(learning_rate=0.005)
>>    model.compile(optimizer=opt,loss='categorical_crossentropy',metrics=['accuracy'])
4. Fit the model for 90 epochs, but changed the batch size to 32. 
trrainX is the numpy array of images. trainy is a list of 8 arrays of targets. 
>>    history = model.fit(trainX, list(trainy), 
                    epochs=90, batch_size=32, 
                    validation_data=(testX, list(testy)), 
                    verbose=1,callbacks=[checkpoint])

I haven't used data augmentation yet. 
The current results look like this: 
- loss: 1.7688 
- Sotfmax_Time_0_loss: 0.1087 - Sotfmax_Time_1_loss: 0.1080 - Sotfmax_Time_2_loss: 0.1090 
- Sotfmax_Time_3_loss: 0.1103 - Sotfmax_Time_4_loss: 0.1121 - Sotfmax_Time_5_loss: 0.1145 
- Sotfmax_Time_6_loss: 0.1191 - Sotfmax_Time_7_loss: 0.1255 

- Sotfmax_Time_0_acc: 0.9616 - Sotfmax_Time_1_acc: 0.9619 - Sotfmax_Time_2_acc: 0.9621 
- Sotfmax_Time_3_acc: 0.9616 - Sotfmax_Time_4_acc: 0.9616 - Sotfmax_Time_5_acc: 0.9607 
- Sotfmax_Time_6_acc: 0.9597 - Sotfmax_Time_7_acc: 0.9592
